src/POData/ObjectModel/CynicDeserialiser.php

```diff
@@ -341,6 +397,7 @@ protected function processLinkFeed(ODataLink &$link, ResourceSet $sourceSet, $so
         assert(isset($bulkResult) && is_array($bulkResult));
 
         for ($i = 0; $i < $numEntries; $i++) {
+            assert($link->expandedResult->entries[$i]->id instanceof KeyDescriptor);
```

Comment:
Enforce that we have actually processed this element's entries and have marked them as such